src/Symfony/Component/Lock/Store/PdoStore.php

```diff
@@ -0,0 +1,378 @@
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * PdoStore is a StoreInterface implementation using a PDO connection.
+ *
+ * Lock metadata are stored in a table. You can use createTable() to initialize
+ * a correctly defined table.
+
+ * CAUTION: This store relies on all client and server nodes to have
+ * synchronized clocks for lock expiry to occur at the correct time.
+ * To ensure locks don't expire prematurely; the ttl's should be set with enough
+ * extra time to account for any clock drift between nodes.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class PdoStore implements StoreInterface
+{
+    private $conn;
+    private $dsn;
+    private $driver;
+    private $table = 'lock_keys';
+    private $idCol = 'key_id';
+    private $tokenCol = 'key_token';
+    private $expirationCol = 'key_expiration';
+    private $username = '';
+    private $password = '';
+    private $connectionOptions = array();
+
+    private $gcProbability;
+    private $initialTtl;
+
+    /**
+     * You can either pass an existing database connection as PDO instance or
+     * a Doctrine DBAL Connection or a DSN string that will be used to
+     * lazy-connect to the database when the lock is actually used.
+     *
+     * List of available options:
+     *  * db_table: The name of the table [default: lock_keys]
+     *  * db_id_col: The column where to store the lock key [default: key_id]
+     *  * db_token_col: The column where to store the lock token [default: key_token]
+     *  * db_expiration_col: The column where to store the expiration [default: key_expiration]
+     *  * db_username: The username when lazy-connect [default: '']
+     *  * db_password: The password when lazy-connect [default: '']
+     *  * db_connection_options: An array of driver-specific connection options [default: array()]
+     *
+     * @param \PDO|Connection|string $connOrDsn     A \PDO or Connection instance or DSN string or null
+     * @param array                  $options       An associative array of options
+     * @param float                  $gcProbability Probability expressed as floating number between 0 and 1 to clean old locks
+     * @param int                    $initialTtl    The expiration delay of locks in seconds
+     *
+     * @throws InvalidArgumentException When first argument is not PDO nor Connection nor string
+     * @throws InvalidArgumentException When PDO error mode is not PDO::ERRMODE_EXCEPTION
+     * @throws InvalidArgumentException When namespace contains invalid characters
+     */
+    public function __construct($connOrDsn, array $options = array(), float $gcProbability = 0.01, int $initialTtl = 300)
+    {
+        if ($gcProbability < 0 || $gcProbability > 1) {
+            throw new InvalidArgumentException(sprintf('"%s" requires gcProbability between 0 and 1, "%f" given.', __CLASS__, $gcProbability));
```

Comment:
Why not use `__METHOD__` like for the next check?

Reply:
Was a partial copy/paste from cache component. 

I switch to __METHOD__ for consistency with other exception in the component